README.md

```diff
@@ -6,34 +6,122 @@
 2. Practice writing functions without side effects
 3. Practice identifying side effects
 
-## Overview
+## What's a pure function?
 
-In many ways, this lesson should be a standard introduction to pure functions:
-students can be presented with, e.g., a cash register that keeps a running
-total:
+![Blue Sky](http://vignette1.wikia.nocookie.net/breakingbad/images/4/43/Season_2_promo_pic_4.jpg/revision/latest?cb=20120617212256)
 
-```javascript
-let total = 0
+_Only 99.1% pure. Amateurs._
 
-function ringUp(cost) {
-  total += cost
+A pure function is a function that:
+ 
+- returns output solely based on its input (the parameters, if there are any);
+- is free from any side effects.
+
+For example:
+
+```js
+const cart = [];
+
+function addToCart(item) {
+  cart.push(item);
 }
 ```
 
-It should be easy to see (with a slightly more detailed example) how keeping a
-total like this can lead to problems for the (imaginary) customers.
+`ringUp()` is _not_ a pure function: the function modifies a variable outside of its own scope. The result wouldn't
```

Comment:
It looks like this is talking about the old example, but the example has changed -- the point stays the same, but we're using `addToCart()` instead of `ringUp()` and `item` instead of `total`.